src/features/courses/course-controller.ts

```diff
@@ -496,16 +532,20 @@ class CourseController {
     };
     
     async createCourse(options: CreateCourseOptions): Promise<Course> {
-        if (options.options.useCurriculum) {
+        if (options.object.curriculumId || options.object.originatingCourseId) {
+            if (options.object.curriculumId && options.object.originatingCourseId) {
+                logger.warn(`Both a curriculum and an originating course were specified. ${options.object.curriculumId}, ${options.object.originatingCourseId}`);
+            }
+
             return useDatabaseTransaction(async () => {
-                // I didn't want this in the transaction, however use strict throws errors if not
-                if (_.isNil(options.object.curriculumId)) {
-                    throw new NotFoundError('Cannot useCurriculum if curriculumId is not given');
-                }
-                const curriculum = await curriculumRepository.getCurriculumById(options.object.curriculumId);
+                
+                // const curriculum = await curriculumRepository.getCurriculumById(options.object.curriculumId);
+                const curriculum = options.object.originatingCourseId ? 
```

Comment:
If curriculum is `Curriculum | Course` we should not be lazy and we should rename the variable